src/utils/projects.ts

```diff
@@ -1,15 +1,15 @@
 import { parseSemester } from "./schema";
 import type { Project } from "./schema";
 
-export interface SemesterProjects {
+export type SemesterProjects = {
   featured?: Project;
   projects: Project[];
-}
+};
 
-export interface ProjectsInfo {
+export type ProjectsInfo = {
   projectMap: Record<string, SemesterProjects>;
   semesters: string[];
-}
+};
```

Comment:
Another personal preference for TS - I like to default to `interface` and only use `type` if I'm making use of union types or generics (i.e. `type Status = 'good' | 'bad'` or `type Result = Promise<...>`.

This is also nonblocking feedback

Reply:
This is a difficult one because there are tradeoffs to both sides since SvelteKit is a bit weird. With interfaces, another type param must be added whenever these interfaces are returned from an endpoint function. See here: https://github.com/sveltejs/kit/issues/5468#issuecomment-1182231341
Also, the RequestHandler type must be imported from the SvelteKit generated types which isn't as much of an issue.

Using a type, this is not necessary. I personally think it looks cleaner when it's implicit, but I'm not entirely sure if this is enough to justify a change. Here's a TS playground that shows roughly why interface typed objects can't be returned from endpoint functions (without the steps in the comment above): [playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAUgygeQHIDUCGAbArtAvFAZ2ACcBLAOwHMoAfWRJBAIwCsIBjYAbgCgeLgEYgDM07aPGTM2nKAG8eASADaAawggAXIRIVKAXW2TUmHLwC+fAUNHioASQC2aShHvlBIsdAVQ-hYnZtIjIqCz5QSAdnVwAVcDx5Hn8AoJ1QynCedgB7ciIoay9xaSMGaQ5gKHw5VO0AckEieqhzKDQCaJc3DxtvXmy8gsiIUvopVkrq+TqoRohm1vbOp274yC4gA)

Reply:
Ah gotcha, since it's a Svelte issue sgtm